packages/dotcom-shared/src/mutators.ts

```diff
@@ -56,25 +56,30 @@ export function createMutators(userId: string) {
 			},
 		},
 		file: {
-			insert: async (tx, file: TlaFile) => {
-				assert(file.ownerId === userId, ZErrorCode.forbidden)
-				await assertNotMaxFiles(tx, userId)
-
-				await tx.mutate.file.insert(file)
-			},
 			insertWithFileState: async (
 				tx,
 				{ file, fileState }: { file: TlaFile; fileState: TlaFileState }
 			) => {
 				assert(file.ownerId === userId, ZErrorCode.forbidden)
 				await assertNotMaxFiles(tx, userId)
+				assert(file.id.match(/^[a-zA-Z0-9_-]+$/), ZErrorCode.bad_request)
+				assert(file.id.length <= 32, ZErrorCode.bad_request)
+				assert(file.id.length >= 16, ZErrorCode.bad_request)
+				assert(file.id === fileState.fileId, ZErrorCode.bad_request)
+				assert(fileState.userId === userId, ZErrorCode.forbidden)
 
 				await tx.mutate.file.upsert(file)
 				await tx.mutate.file_state.upsert(fileState)
 			},
 			deleteOrForget: async (tx, file: TlaFile) => {
 				await tx.mutate.file_state.delete({ fileId: file.id, userId })
 				if (file?.ownerId === userId) {
+					if (tx.location === 'server') {
+						// todo: use a sql trigger for this like we do for setting shared to false
+						await tx.dbTransaction.query(`delete from public.file_state where "fileId" = $1`, [
+							file.id,
+						])
+					}
```

Comment:
drive-by fix to clean up all files states after soft deleting a file. we definitely should be using a trigger for this but it can wait for a follow up.